### PR TITLE
feat: replace AnimatedTitle with SquigglyTitle

### PR DESCRIPTION
**Pull request by 0xShishigami**
## ISSUE
Using video for animated titles was marked with poor performance score by _Lighthouse_.

[Landing performance scoring](https://googlechrome.github.io/lighthouse/scorecalc/#FCP=1083&LCP=5044&TBT=240&CLS=0&SI=2220&TTI=5131&device=mobile&version=12.1.0) (77)
Metric: Largest Contentful Paint ([LCP](https://web.dev/articles/lcp))
Device: Mobile

Screenshot 2024-09-02 at 11 47 16 AM

## SOLUTION
- Implement [SVG filter primitives
](https://drafts.fxtf.org/filter-effects/#FilterPrimitivesOverview)
- Browser compatibiliy: [Full support](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/filter#browser_compatibility)

## DONE
- Add `SquigglyTitle` component
- Replace AnimatedTitle with the new component
- Perfomance scoring 77 -> 87

Screenshot 2024-09-02 at 11 29 35 AM

## PREVIEW
CURRENT  |  PROPOSED

Screenshot 2024-09-02 at 10 57 02 AM

### CODE REVIEW

File: src/pages/creations/index.tsx [hunk 1]
```diff
@@ -33,9 +30,16 @@ export default function Creations() {
               <BG_3 type='2' align='right' />
             </BackgroundContainer>
             <HeroDivider>
-              <TitleVideo>
-                <AnimatedTitle chromeSrc={VIDEO_CHROME} safariSrc={VIDEO_SAFARI} />
-              </TitleVideo>
+              <SquigglyTitle
```
Comment (0xArdy) [inline src/pages/creations/index.tsx:33]: this title should be more centered in desktop and mobile views

before, 
image
after,
image

File: src/pages/creations/index.tsx [hunk 2]
```diff
@@ -33,9 +30,16 @@ export default function Creations() {
...
+                text={`WHAT'S COOKING?`}
```
Comment (0xArdy) [inline src/pages/creations/index.tsx:34]:
```suggestion
                text="WHAT'S COOKING?"
```

File: src/pages/ethos/index.tsx [hunk 3]
```diff
@@ -29,9 +26,16 @@ export default function Ethos() {
       <ContentContainer>
         <PageContent>
           <HeroDivider>
-            <STitleContainer>
-              <AnimatedTitle chromeSrc={VIDEO_CHROME} safariSrc={VIDEO_SAFARI} />
-            </STitleContainer>
+            <SquigglyTitle
+              text={`WE'RE ALL MAD HERE`}
```
Comment (0xArdy) [inline src/pages/ethos/index.tsx:30]:
```suggestion
              text="WE'RE ALL MAD HERE"
```

File: src/pages/insights/index.tsx [hunk 4]
```diff
@@ -27,7 +25,16 @@ export default function Insights() {
         <PageContainer>
           <Title>
             <BlogTitleContainer>
-              <AnimatedTitle chromeSrc={VIDEO_CHROME} safariSrc={VIDEO_SAFARI} />
+              <SquigglyTitle
+                text={`WHAT'S NEW?`}
```
Comment (0xArdy) [inline src/pages/insights/index.tsx:29]:
```suggestion
                text="WHAT'S NEW?"
```

File: src/pages/landing/HeroSection.tsx [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
+import { Section, MOBILE_MAX_WIDTH, SquigglyTitle } from '~/components';
 
 const HeroSection = ({ ...props }) => {
   return (
     <>
       <StyledHeroSection full backgroundImage='/img/hero/hero-bg.jpg' {...props}>
         <TitleContainer>
-          <AnimatedTitle chromeSrc={VIDEO_CHROME} safariSrc={VIDEO_SAFARI} />
+          <SquigglyTitle
+            text={`THE LARGEST CORE DEVELOPMENT\nGROUP IN WEB3`}
```
Comment (0xArdy) [inline src/pages/landing/HeroSection.tsx:11]:
```suggestion
            text="THE LARGEST CORE DEVELOPMENT\nGROUP IN WEB3"
```

**Review by 0xArdy — COMMENTED** [review]
great implementation! 🔥 

i left just a few comments

File: src/pages/landing/HeroSection.tsx (hunk 5 above)
Comment (0xShishigami) [reply src/pages/landing/HeroSection.tsx:11]:
```suggestion
            text={`THE LARGEST CORE DEVELOPMENT<br/>GROUP IN WEB3`}
```

Template string is necessary for breaking lines using `\n`.

We can use `<br />` instead and `dangerouslySetInnerHTML` into SquigglyTitle component.

Wdyt?

File: src/pages/creations/index.tsx (hunk 1 above)
Comment (0xShishigami) [reply src/pages/creations/index.tsx]: [6697cf9](https://github.com/defi-wonderland/web/pull/110/commits/6697cf92e720fffa1e14803fa4c31bec9226dd83)

File: src/pages/creations/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xShishigami) [reply src/pages/creations/index.tsx]:
```suggestion
            text="WHAT'S COOKING?"
```

ESLint said nothing but ERROR haha maybe can just switch to double quotes for these cases.
ESLint is ok whit that

**Review by 0xArdy — COMMENTED** [review]
"WHAT'S COOKING?" text should be bigger in mobile

old:
image
new:
image

File: src/pages/creations/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xArdy) [reply src/pages/creations/index.tsx]: oh okay, so leave it as `"WHAT'S COOKING?"`

File: src/pages/landing/HeroSection.tsx (hunk 5 above)
Comment (0xArdy) [reply src/pages/landing/HeroSection.tsx:11]: > Template string is necessary for breaking lines using \n.
got it. so leave it as it is 😁

File: src/pages/creations/index.tsx (hunk 2 above)
Comment (0xShishigami) [reply src/pages/creations/index.tsx]: https://github.com/defi-wonderland/web/pull/110/commits/2fd1d6b5d29c14f91d054f5adda08eb7ce0cf621

File: src/pages/ethos/index.tsx (hunk 3 above)
Comment (0xShishigami) [reply src/pages/ethos/index.tsx]: https://github.com/defi-wonderland/web/pull/110/commits/2fd1d6b5d29c14f91d054f5adda08eb7ce0cf621

File: src/pages/insights/index.tsx (hunk 4 above)
Comment (0xShishigami) [reply src/pages/insights/index.tsx]: https://github.com/defi-wonderland/web/pull/110/commits/2fd1d6b5d29c14f91d054f5adda08eb7ce0cf621

File: src/pages/creations/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xShishigami) [reply src/pages/creations/index.tsx]: https://github.com/defi-wonderland/web/pull/110/commits/2fd1d6b5d29c14f91d054f5adda08eb7ce0cf621

**Comment by 0xShishigami** [comment]
> "WHAT'S COOKING?" text should be bigger in mobile
> 

@0xArdy 

https://github.com/defi-wonderland/web/pull/110/commits/09fb880ce2b30ad26dffe2bad58d4a7c430ac792

**Review by 0xArdy — APPROVED** [review]

**Review by 0xtiti — DISMISSED** [review]
🔥

File: src/components/SquigglyTitle.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+              x='0%'
+              y='0%'
+              width='100%'
+              height='100%'
+              result='displacementMap'
+            >
+              <animate attributeName='scale' values='5;15;5;' dur='2' repeatCount='indefinite' begin='0' />
+            </feDisplacementMap>
+          </filter>
+          {/* Squiggly filter mobile */}
```
Comment (Onizuka-wl) [inline src/components/SquigglyTitle.tsx:69]: are the comments necessary given that the ID already indicates whether it's for mobile or desktop?

**Review by Onizuka-wl — DISMISSED** [review]

File: src/components/SquigglyTitle.tsx (hunk 6 above)
Comment (0xShishigami) [reply src/components/SquigglyTitle.tsx:69]: I think it helps not to be scared with a lot of uncommon tags 😅 and quickly answer why the filter is "duplicated".
In a next PR could be simplified to just one filter and modify the attributes with JS for mobile.

**Comment by 0xLaruku** [comment]
@0xShishigami 
Looks fine in Windows Chrome and Android Chrome.
But the titles are not completely displayed/doesn't display at all, for macOS Safari and iOS Safari:

**iOS Safari (Iphone 14 PRO Max)**
[Ethos](https://drive.google.com/file/d/1rVvKy6x0ovLA10UXwg9EpPeMtjAsMVCi/view?usp=sharing)
[Creations](https://drive.google.com/file/d/1KlX2NxMDWq8ioapC4z4h7DiTQuvRzzpV/view?usp=sharing)
[Insights](https://drive.google.com/file/d/14FDWWTkI1Tot6K81vdeBn-R6_7yiHbCw/view?usp=sharing)

**macOS**
[Home ](https://drive.google.com/file/d/1IOPOElJuVWkSt_TDBI3_n2z2bdPc1Nyt/view?usp=sharing)(part of the development's T is cut)
[Ethos](https://drive.google.com/file/d/1vD3D3acqj_s55R4EKBdbqLlvmlGRVvOw/view?usp=sharing)
[Creations](https://drive.google.com/file/d/13ar-dN4YrJ6aQ2sg9n1gqp8eFTM_tY0J/view?usp=sharing)
[Insights](https://drive.google.com/file/d/1IfdliUec8tHcKebkEO4EFfu1vtHOSPZc/view?usp=sharing)

Also, in desktop the text looks a bit pixelated to me: [screenshot is in Chrome (Windows)](https://drive.google.com/file/d/1wPbXtanmrfCCrpnu4m6X7J1aQ1JuySzf/view?usp=sharing)

**Comment by 0xShishigami** [comment]
> @0xShishigami Looks fine in Windows Chrome and Android Chrome. But the titles are not completely displayed/doesn't display at all, for macOS Safari and iOS Safari:
> 
> **iOS Safari (Iphone 14 PRO Max)** [Ethos](https://drive.google.com/file/d/1rVvKy6x0ovLA10UXwg9EpPeMtjAsMVCi/view?usp=sharing) [Creations](https://drive.google.com/file/d/1KlX2NxMDWq8ioapC4z4h7DiTQuvRzzpV/view?usp=sharing) [Insights](https://drive.google.com/file/d/14FDWWTkI1Tot6K81vdeBn-R6_7yiHbCw/view?usp=sharing)
> 
> **macOS** [Home ](https://drive.google.com/file/d/1IOPOElJuVWkSt_TDBI3_n2z2bdPc1Nyt/view?usp=sharing)(part of the development's T is cut) [Ethos](https://drive.google.com/file/d/1vD3D3acqj_s55R4EKBdbqLlvmlGRVvOw/view?usp=sharing) [Creations](https://drive.google.com/file/d/13ar-dN4YrJ6aQ2sg9n1gqp8eFTM_tY0J/view?usp=sharing) [Insights](https://drive.google.com/file/d/1IfdliUec8tHcKebkEO4EFfu1vtHOSPZc/view?usp=sharing)
> 
> Also, in desktop the text looks a bit pixelated to me: [screenshot is in Chrome (Windows)](https://drive.google.com/file/d/1wPbXtanmrfCCrpnu4m6X7J1aQ1JuySzf/view?usp=sharing)

Should be displayed correctly now! https://github.com/defi-wonderland/web/pull/110/commits/208ae42aeae5c2783809e73d4b03cdf2b46f9156

Regarding the pixelated result: sadly it's the "expected" behavior because we are doing bitmap manipulations.

**Review by 0xArdy — DISMISSED** [review]

**Comment by 0xLaruku** [comment]
> > @0xShishigami Looks fine in Windows Chrome and Android Chrome. But the titles are not completely displayed/doesn't display at all, for macOS Safari and iOS Safari:
> > **iOS Safari (Iphone 14 PRO Max)** [Ethos](https://drive.google.com/file/d/1rVvKy6x0ovLA10UXwg9EpPeMtjAsMVCi/view?usp=sharing) [Creations](https://drive.google.com/file/d/1KlX2NxMDWq8ioapC4z4h7DiTQuvRzzpV/view?usp=sharing) [Insights](https://drive.google.com/file/d/14FDWWTkI1Tot6K81vdeBn-R6_7yiHbCw/view?usp=sharing)
> > **macOS** [Home ](https://drive.google.com/file/d/1IOPOElJuVWkSt_TDBI3_n2z2bdPc1Nyt/view?usp=sharing)(part of the development's T is cut) [Ethos](https://drive.google.com/file/d/1vD3D3acqj_s55R4EKBdbqLlvmlGRVvOw/view?usp=sharing) [Creations](https://drive.google.com/file/d/13ar-dN4YrJ6aQ2sg9n1gqp8eFTM_tY0J/view?usp=sharing) [Insights](https://drive.google.com/file/d/1IfdliUec8tHcKebkEO4EFfu1vtHOSPZc/view?usp=sharing)
> > Also, in desktop the text looks a bit pixelated to me: [screenshot is in Chrome (Windows)](https://drive.google.com/file/d/1wPbXtanmrfCCrpnu4m6X7J1aQ1JuySzf/view?usp=sharing)
> 
> Should be displayed correctly now! [208ae42](https://github.com/defi-wonderland/web/commit/208ae42aeae5c2783809e73d4b03cdf2b46f9156)
> 
> Regarding the pixelated result: sadly it's the "expected" behavior because we are doing bitmap manipulations.
> 
> CC @0xArdy @0xtiti

✅ Fixed for iOS Safari, evidence [here](https://drive.google.com/drive/folders/1MXtIn1oxmlBgw9XCvM6ItVt_tw7msqej?usp=drive_link)

❌ Still issues in the desktop version for macOS Safari. The titles are displayed completely now, but they stop moving right after navigating to the sections, they starts moving only after refreshing the page: Video [here](https://drive.google.com/file/d/1sSkibivP0XlgpWHDhfDLfw5p7YJG--oD/view?usp=sharing)

**Comment by 0xShishigami** [comment]
> > > @0xShishigami Looks fine in Windows Chrome and Android Chrome. But the titles are not completely displayed/doesn't display at all, for macOS Safari and iOS Safari:
> > > **iOS Safari (Iphone 14 PRO Max)** [Ethos](https://drive.google.com/file/d/1rVvKy6x0ovLA10UXwg9EpPeMtjAsMVCi/view?usp=sharing) [Creations](https://drive.google.com/file/d/1KlX2NxMDWq8ioapC4z4h7DiTQuvRzzpV/view?usp=sharing) [Insights](https://drive.google.com/file/d/14FDWWTkI1Tot6K81vdeBn-R6_7yiHbCw/view?usp=sharing)
> > > **macOS** [Home ](https://drive.google.com/file/d/1IOPOElJuVWkSt_TDBI3_n2z2bdPc1Nyt/view?usp=sharing)(part of the development's T is cut) [Ethos](https://drive.google.com/file/d/1vD3D3acqj_s55R4EKBdbqLlvmlGRVvOw/view?usp=sharing) [Creations](https://drive.google.com/file/d/13ar-dN4YrJ6aQ2sg9n1gqp8eFTM_tY0J/view?usp=sharing) [Insights](https://drive.google.com/file/d/1IfdliUec8tHcKebkEO4EFfu1vtHOSPZc/view?usp=sharing)
> > > Also, in desktop the text looks a bit pixelated to me: [screenshot is in Chrome (Windows)](https://drive.google.com/file/d/1wPbXtanmrfCCrpnu4m6X7J1aQ1JuySzf/view?usp=sharing)
> > 
> > 
> > Should be displayed correctly now! [208ae42](https://github.com/defi-wonderland/web/commit/208ae42aeae5c2783809e73d4b03cdf2b46f9156)
> > Regarding the pixelated result: sadly it's the "expected" behavior because we are doing bitmap manipulations.
> > CC @0xArdy @0xtiti
> 
> ✅ Fixed for iOS Safari, evidence [here](https://drive.google.com/drive/folders/1MXtIn1oxmlBgw9XCvM6ItVt_tw7msqej?usp=drive_link)
> 
> ❌ Still issues in the desktop version for macOS Safari. The titles are displayed completely now, but they stop moving right after navigating to the sections, they starts moving only after refreshing the page: Video [here](https://drive.google.com/file/d/1sSkibivP0XlgpWHDhfDLfw5p7YJG--oD/view?usp=sharing)

Hate Safari so much.
After some deep research found that it has a lot of bugs related to SVG filters.
I have a working version using the `<text>` tag, but have yet to redo the resizing logic for responsiveness.

Screenshot 2024-09-13 at 8 39 50 PM

**Comment by 0xShishigami** [comment]
> Title animation was disable for Safari and wrappers of Safari (like Chrome for iOS devices).

**Mac OS:**
Safari and Chrome

**iPhone 16 Plus iOS 18 (simulated with xCode):**
Safari

**iPhone 14 Pro iOS 17.5:**
Safari and Chrome

**Review by 0xArdy — APPROVED** [review]

**Review by 0xtiti — APPROVED** [review]

**Pull request merged**